### PR TITLE
REF: avoid _internal_get_values in json get_values

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @WillAyd 

As mentioned in #32731, this changes the behavior for CategoricalIndex[dt64tz] and Series with that for its index.  Suggestions for where tests for that should go?

Note: the Categorical[dt64tz] cases still do a trip through object-dtype; I don't want to futz with that in C-space.

### CODE REVIEW

**Comment by WillAyd** [comment]
> Suggestions for where tests for that should go?

Pretty much everything right now goes into `test_pandas`; there's a PR to start fixturing / splitting things that I promise I'll get back to...but for now just place there

File: pandas/_libs/src/ujson/python/objToJSON.c [hunk 1]
```diff
@@ -222,13 +222,19 @@ static PyObject *get_values(PyObject *obj) {
 
     PRINTMARK();
 
-    if (PyObject_HasAttrString(obj, "_internal_get_values")) {
+    if (PyObject_TypeCheck(obj, cls_index) || PyObject_TypeCheck(obj, cls_series)) {
```
Comment (WillAyd) [inline pandas/_libs/src/ujson/python/objToJSON.c]: I guess will see on test cases, but is this really required? AFAICT on master `_internal_get_values` and `values` return the same for a dt64tz array:

```python
>>> dts = pd.date_range(start="2020-01-01", freq="D", periods=3, tz="US/Pacific")
>>> dts._internal_get_values()
array(['2020-01-01T08:00:00.000000000', '2020-01-02T08:00:00.000000000',
       '2020-01-03T08:00:00.000000000'], dtype='datetime64[ns]')
>>> dts.values
array(['2020-01-01T08:00:00.000000000', '2020-01-02T08:00:00.000000000',
       '2020-01-03T08:00:00.000000000'], dtype='datetime64[ns]')
```

**Review by WillAyd — COMMENTED** [review]
Nice simplification!

File: pandas/_libs/src/ujson/python/objToJSON.c (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/src/ujson/python/objToJSON.c]: its Categorical[dt64tz] and Sparse that make the extra `__array__` call necessary

**Comment by jreback** [comment]
lgtm. happy if you guys are.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**